add optional system schema authorization (#11720)
apache/druid: docs/operations/security-user-auth.md

At the center of the Druid user authentication and authorization model are _resources_ and _actions_. A resource is something that authenticated users are trying to access or modify. An action is something that users are trying to do. 

> There are four resource types:

Author: heh this is actually a lie, there is also `VIEW`, but since there is no view implementation I don't think we need to call it out yet

> There are four resource types:

Reviewer: ```suggestion
Druid uses the following resource types:
```
avoid specific number

* DATASOURCE &ndash; Each Druid table (i.e., `tables` in the `druid` schema in SQL) is a resource.
* CONFIG &ndash; Configuration resources exposed by the cluster components. 
* STATE &ndash; Cluster-wide state resources.

> * SYSTEM_TABLE &ndash; if `druid.sql.planner.authorizeSystemTablesDirectly` is enabled, then system tables (the `sys` schema in SQL) are authorized as this type of resource.

Reviewer: ```suggestion
* SYSTEM_TABLE &ndash; if `druid.sql.planner.authorizeSystemTablesDirectly` is enabled, then Druid authorizes system tables,`sys` schema in SQL, using this resource type.
```
